pkg/common/genericSet.go

```diff
@@ -51,3 +51,16 @@ func (s GenericSet[T]) IsIntersect(s2 GenericSet[T]) bool {
 	}
 	return false
 }
+
+// /////////////////////////////////////////////////////////////////
+// AnyMapEntry() return one arbitrary entry of a map.
+// this func is not related to genericSet,
+// todo: consider moving to another file
+// todo: AnyMapEntry() for GenericSet, and use it
+// /////////////////////////////////////////////////////////
+func AnyMapEntry[K comparable, V any](m map[K]V) (k K, v V) {
```

Comment:
My editor hates this line, marking it as ERROR:
"Invalid map key type: comparison operators == and != must be fully defined for the key type"

Reply:
K is comparable, so I do not understand this error.
try to upgrade  to golang 21. I had generic issues when my golang was set to older version

Reply:
I have golang `1.21.5`

Reply:
this error does not make sense.
looks like a golang error that was solved 2 years ago.
you can see same issue here (should be solved):
https://youtrack.jetbrains.com/issue/GO-12615/Go-1.18-GenericsInvalid-map-key-type-comparison-operators-and-must-be-fully-defined-for-key-types

Reply:
weird, my version is 1.3.4